File: packages/web/src/types/report.ts
export type Verdict = 'green' | 'yellow' | 'red'

export type SignalCategory =
  | 'team'
  | 'legal'
  | 'engineering'
  | 'hiring'
  | 'customer'
  | 'financials'
  | 'founder'
  | 'product'

export type SignalSentiment = 'positive' | 'negative' | 'neutral'

export interface Signal {
  category: SignalCategory
  title: string
  description: string
  sentiment: SignalSentiment
  source: string
  date_detected: string | null
  weight: number
}

export interface IntelligenceReport {
  company_name: string
  generated_at: string
  verdict: Verdict
  confidence: number
  summary: string
  signals: Signal[]
  sources_used: string[]
  raw_chunks_count: number
}

File: packages/rag/schema/report.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

from .signal import Signal


class IntelligenceReport(BaseModel):
    company_name: str
    generated_at: datetime
    verdict: Literal["green", "yellow", "red"]
    confidence: float
    summary: str
    signals: list[Signal]
    sources_used: list[str]
    raw_chunks_count: int = 0
